strawberryfields/api/devicespec.py

```diff
@@ -97,17 +99,35 @@ def gate_parameters(self):
 
         return gate_parameters
 
+    def layout_is_formatted(self):
+        """bool: Whether the device layout is formatted or not."""
+        p = re.compile(r"{{\w*}}")
+        return not bool(p.search(self.layout))
+
+    def fill_template(self, program):
+        """Fill template with parameter values from a program"""
+        if self.layout_is_formatted():
+            return
+
+        if program.type == "tdm" and program.timebins:
+            self._spec["layout"] = self._spec["layout"].format(
+                target=self.target, tm=program.timebins
+            )
+        else:
+            # TODO: update when `self._spec["layout"]` is returned as an unformatted string
+            raise NotImplementedError("Formatting not required or supported for non-TDM programs.")
```

Comment:
Please add this to the clubhouse backlog so that we don't forget!

Reply:
[Backlog ticket added!](https://app.clubhouse.io/xanaduai/story/2813/update-fill-template-to-support-non-tdm-layouts)